src/Dfc.CourseDirectory.Web/Controllers/CourseSummaryController.cs

```diff
@@ -118,45 +118,20 @@ public IActionResult Index(Guid? courseId, Guid? courseRunId)
                 }
             }
 
-            if(courseRun.Regions != null)
+            if (courseRun.Regions != null)
             {
-                var allRegions = _courseService.GetRegions().RegionItems;
-                var regions = GetRegions().RegionItems.Select(x => x.Id);
-                vm.Regions = FormattedRegionsByIds(allRegions, courseRun.Regions);
+                var allRegions = _courseService.GetRegions();
+
+                vm.Regions = allRegions.RegionItems
+                    .Select(r => new { r.Id, r.RegionName })
+                    .Concat(allRegions.RegionItems
+                        .SelectMany(r => r.SubRegion)
+                        .Select(r => new { r.Id, RegionName = r.SubRegionName }))
+                    .Where(r => allRegions.SubRegionsDataCleanse(courseRun.Regions.ToList()).Contains(r.Id))
```

Comment:
I think that `ToList` going to be run for every record. You might want to extract it to a variable or something.

Reply:
Good spot - fixed.